### ISSUE TITLE
Vagrant almost works

### ISSUE THREAD

**Issue by cantino** [cantino/huginn#740]
It launches and provisions with chef, but then Huginn doesn't start.  Should role[huginn_development] be starting it automatically? @alias1

**Comment by mrjcleaver** [comment]
I concur. Here's me finishing a vagrant deploy, reboot and finding the MySQL server had gone AWOL.

````
==> default: [2015-04-22T20:00:59+00:00] INFO: git[/home/huginn/huginn] checked out branch: master onto: deploy reference: 3fc2c95be1e0c87cd877f3a5f3a63e0244070b14
==> default: [2015-04-22T20:01:35+00:00] INFO: gem_package[bundle] installed bundle at 0.0.1
==> default: [2015-04-22T20:01:35+00:00] INFO: bash[Setting huginn user with NOPASSWD option] ran successfully
==> default: [2015-04-22T20:31:37+00:00] INFO: bash[huginn dependencies] ran successfully
==> default: [2015-04-22T20:31:46+00:00] INFO: bash[huginn has been installed and will start in a minute] ran successfully
==> default: [2015-04-22T20:31:47+00:00] INFO: Chef Run complete in 3046.961883177 seconds
==> default: [2015-04-22T20:31:47+00:00] INFO: Skipping removal of unused files from the cache
==> default: [2015-04-22T20:31:47+00:00] INFO: Running report handlers
==> default: [2015-04-22T20:31:47+00:00] INFO: Report handlers complete
martincleaver@MartinCaversMBP:~/SoftwareDevelopment/huginn/deployment 16:38:44 596$ vagrant ssh
Welcome to Ubuntu 12.04 LTS (GNU/Linux 3.2.0-23-generic x86_64)

 * Documentation:  https://help.ubuntu.com/

183 packages can be updated.
97 updates are security updates.

New release '14.04.2 LTS' available.
Run 'do-release-upgrade' to upgrade to it.



Welcome to your Vagrant-built virtual machine.
Last login: Fri Sep 14 06:23:18 2012 from 10.0.2.2
vagrant@precise64:~$ sudo su -
root@precise64:~# ls /etc/init.d/
apparmor                     killprocs                    pppd-dns                     statd-mounting
atd                          .legacy-bootordering         procps                       stop-bootlogd
bootlogd                     module-init-tools            rc                           stop-bootlogd-single
console-setup                mysql                        rc.local                     sudo
cron                         networking                   rcS                          udev
dbus                         network-interface            README                       udev-fallback-graphics
dmesg                        network-interface-container  reboot                       udev-finish
dns-clean                    network-interface-security   resolvconf                   udevmonitor
friendly-recovery            ntp                          rpcbind-boot                 udevtrigger
grub-common                  ondemand                     rsync                        ufw
gssd                         plymouth                     rsyslog                      umountfs
halt                         plymouth-log                 sendsigs                     umountnfs.sh
hostname                     plymouth-splash              setvtrgb                     umountroot
hwclock                      plymouth-stop                single                       urandom
hwclock-save                 plymouth-upstart-bridge      skeleton                     vboxadd
idmapd                       portmap                      ssh                          vboxadd-service
irqbalance                   portmap-wait                 statd                        vboxadd-x11
root@precise64:~# start huginn
start: Unknown job: huginn
root@precise64:~# ls /home/huginn/
.bash_logout  .bashrc       huginn/       .profile      
root@precise64:~# ls /home/huginn/huginn/
app/                 config.ru            .env.example         lib/                 Procfile.CF          .travis.yml
app.json             data/                Gemfile              LICENSE              public/              vendor/
bin/                 db/                  Gemfile.lock         log/                 Rakefile             VERSION
.buildpacks          deployment/          .git/                manifest.yml.sample  README.md            
Capfile              doc/                 .gitignore           media/               script/              
CHANGES.md           docker/              .graphviz            .openshift/          spec/                
config/              .env                 Guardfile            Procfile             tmp/                 
root@precise64:~# ls /home/huginn/huginn/bin/
bundle             rails              rspec              setup_heroku       spring             twitter_stream.rb
decrypt_backup.rb  rake               schedule.rb        setup_openshift    threaded.rb        
root@precise64:~# telnet localhost 80
Trying 127.0.0.1...
telnet: Unable to connect to remote host: Connection refused
root@precise64:~# ls
root@precise64:~# pwd
/root
root@precise64:~# cd -
-su: cd: OLDPWD not set
root@precise64:~# cd ~huginn/
root@precise64:/home/huginn# ls
huginn
root@precise64:/home/huginn# cd huginn/
root@precise64:/home/huginn/huginn# ls
app       Capfile     config.ru  deployment  Gemfile       lib      manifest.yml.sample  Procfile.CF  README.md  tmp
app.json  CHANGES.md  data       doc         Gemfile.lock  LICENSE  media                public       script     vendor
bin       config      db         docker      Guardfile     log      Procfile             Rakefile     spec       VERSION
root@precise64:/home/huginn/huginn# foreman start
21:29:24 web.1  | started with pid 6806
21:29:24 jobs.1 | started with pid 6808
21:29:31 jobs.1 | Warning: You're using Rubygems 1.8.11 with Spring. Upgrade to at least Rubygems 2.1.0 and run `gem pristine --all` for better startup performance.
21:29:31 web.1  | Warning: You're using Rubygems 1.8.11 with Spring. Upgrade to at least Rubygems 2.1.0 and run `gem pristine --all` for better startup performance.
21:29:39 jobs.1 | /var/lib/gems/1.9.1/gems/spring-1.3.3/lib/spring/client/run.rb:118:in `connect_to_application': Error connecting to Spring server (RuntimeError)
21:29:39 jobs.1 | 	from /var/lib/gems/1.9.1/gems/spring-1.3.3/lib/spring/client/run.rb:61:in `run'
21:29:39 jobs.1 | 	from /var/lib/gems/1.9.1/gems/spring-1.3.3/lib/spring/client/run.rb:37:in `warm_run'
21:29:39 jobs.1 | 	from /var/lib/gems/1.9.1/gems/spring-1.3.3/lib/spring/client/run.rb:26:in `call'
21:29:39 jobs.1 | 	from /var/lib/gems/1.9.1/gems/spring-1.3.3/lib/spring/client/command.rb:7:in `call'
21:29:39 jobs.1 | 	from /var/lib/gems/1.9.1/gems/spring-1.3.3/lib/spring/client/rails.rb:23:in `call'
21:29:39 jobs.1 | 	from /var/lib/gems/1.9.1/gems/spring-1.3.3/lib/spring/client/command.rb:7:in `call'
21:29:39 jobs.1 | 	from /var/lib/gems/1.9.1/gems/spring-1.3.3/lib/spring/client.rb:26:in `run'
21:29:39 jobs.1 | 	from /var/lib/gems/1.9.1/gems/spring-1.3.3/bin/spring:48:in `<top (required)>'
21:29:39 jobs.1 | 	from /var/lib/gems/1.9.1/gems/spring-1.3.3/lib/spring/binstub.rb:11:in `load'
21:29:39 jobs.1 | 	from /var/lib/gems/1.9.1/gems/spring-1.3.3/lib/spring/binstub.rb:11:in `<top (required)>'
21:29:39 jobs.1 | 	from /home/huginn/huginn/bin/spring:13:in `require'
21:29:39 jobs.1 | 	from /home/huginn/huginn/bin/spring:13:in `<top (required)>'
21:29:39 jobs.1 | 	from bin/rails:3:in `load'
21:29:39 jobs.1 | 	from bin/rails:3:in `<main>'
21:29:39 jobs.1 | exited with code 1
21:29:39 system | sending SIGTERM to all processes
21:29:39 web.1  | terminated by SIGTERM
root@precise64:/home/huginn/huginn# reboot
root@precise64:/home/huginn/huginn# 
Broadcast message from vagrant@precise64
	(/dev/pts/1) at 21:31 ...

The system is going down for reboot NOW!
Connection to 127.0.0.1 closed by remote host.
Connection to 127.0.0.1 closed.
````

Then:
````
martincleaver@MartinCaversMBP:~/SoftwareDevelopment/huginn/deployment 17:32:07 597$ vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Checking if box 'hashicorp/precise64' is up to date...
==> default: VirtualBox VM is already running.
martincleaver@MartinCaversMBP:~/SoftwareDevelopment/huginn/deployment 17:37:05 598$ lynx localhost 3000
Looking up  'localhost' first

Looking up 3000 first
Looking up 3000
Making HTTP connection to 3000
Alert!: Unable to connect to remote host.

lynx: Can't access startfile http://3000/
martincleaver@MartinCaversMBP:~/SoftwareDevelopment/huginn/deployment 17:41:24 599$ lynx localhost:3000


Exiting via interrupt: 2

````
martincleaver@MartinCaversMBP:~/SoftwareDevelopment/huginn/deployment 17:47:08 600$ vagrant ssh
Welcome to Ubuntu 12.04 LTS (GNU/Linux 3.2.0-23-generic x86_64)

 * Documentation:  https://help.ubuntu.com/

183 packages can be updated.
97 updates are security updates.

New release '14.04.2 LTS' available.
Run 'do-release-upgrade' to upgrade to it.



Welcome to your Vagrant-built virtual machine.
Last login: Wed Apr 22 21:21:02 2015 from 10.0.2.2
````

I note:
````
  SQL (7796.4ms)  INSERT INTO `delayed_jobs` (`handler`, `priority`, `run_at`, `created_at`, `updated_at`) VALUES ('--
- !ruby/object:Delayed::PerformableMethod\nobject: !ruby/class \'Agent\'\nmethod_name: :receive!\nargs: []\n', 10, '2015-04-22 20:4
7:53', '2015-04-22 20:48:01', '2015-04-22 20:48:01')
   (11170.8ms)  COMMIT
Mysql2::Error: Lost connection to MySQL server during query: COMMIT
   (19923.2ms)  ROLLBACK
Mysql2::Error: MySQL server has gone away: ROLLBACK
````

So in summary I did:
````
    1  start huginn
    2  telnet localhost 80
    3  cd
    4  pwd
    5  cd -
    6  cd ~huginn/
    7  ls
    8  cd huginn/
    9  ls
   10  foreman start
   11  reboot
   12  cd ~huginn/
   13  ls
   14  cd huginn/
   15* 
   16  foreman start
   17  bg
   18  lynx http://localhost
   19  ls
   20  more log/development.log 
````

I'll try a remote deployment. I don't see errors but I can imagine this is memory hungry.

**Comment by mrjcleaver** [comment]
https://github.com/cantino/huginn/blob/master/deployment/site-cookbooks/huginn_production/recipes/default.rb#L106 is a useful reference point.

**Comment by alias1** [comment]
https://github.com/alias1/huginn-chef May also be useful/relevant

**Comment by cantino** [comment]
@dsander, does #872 make any effort to update or remove Vagrant?

**Comment by elijah** [comment]
I'd like suggestions on how to refactor the docs to account for the outside
work that I'm doing to try to remove this cruft from the huginn repo.

Do I write docs and make a PR for them before things are completely ready,
do I PR the work and then revise the docs, or somewhere in between?

I don't want anybody left with bad docs, but neither do I want to have
things in a more-broken state, not even for a few hours....

Writing glue this morning to mostly automate the setup that's in the docs
now.  I think the whole page referenced in the wiki for vagrant and
chef-solo instructions can get a lot shorter, soon...

--e

**Comment by elijah** [comment]
I just added a section to the docs to reference the external repos, as a
third section of the page at
https://github.com/cantino/huginn/wiki/Deploying-huginn-on-any-server-or-virtualbox-using-chef-solo-and-or-vagrant

There are a few more things I want to work out in those repos - hopefully I
can make some more headway today.

--e

**Comment by elijah** [comment]
I managed to fix up the broken vagrant bits today, to the extent that I can
actually do a 'vagrant up' in the vagrant-huginn checkout now and get
things to run.

The big mind-twister was that the mysql, database, and mysql_chef_gem
cookbooks underwent a massive refactor a year or so back.  You used to be
able to just include mysql::database - that recipe just doesn't exist
anymore, and was replaced with a bunch of other resources.  It took me a
while to get the dependencies right - but I did.  In cookbook-huginn,
there's now a recipes/database.rb file - right now it's filled in with some
Very Silly default values (like the password is still... 'password'....)
and needs to have some more work to make it a little more database neutral
and let us switch postgres/mysql in and out.

Same deal with rvm and rbenv, etc - huginn::default needs to use 'em
instead of the installs it does now.  :-)

A bunch of other things have become obvious too - like the need to rewrite
the env.example as an ERB template, so I can just override settings like
the DB password from Vagrant/elsewhere.

Happy to have made real serious progress today.  My local copy of huginn
works now, which is pretty cool.  :)

best,

--elijah

**Comment by dsander** [comment]
@cantino I was my initial plan to remove the vagrant/chef docs but since @elijah is working on it, I think it should be his call.

@elijah That sounds awesome! I will try to give it a spin on the weekend.

**Comment by elijah** [comment]
What's there now is OK, but probably needs to be gone over very very
thoroughly for errors, and generally modernized.  [It tells you a bunch of
things that are certainly good to do, but skims over details that might
impact someone.]  It's sort of on my should-do list, but it takes a lot of
energy to rewrite documentation.  ;-)

I stuck a Makefile in vagrant-huginn that has a default target that I think
'mostly' installs the bits needed to get things going;  it's pretty
OSX-centric at the moment, and is missing pieces, and could generally be
made much better for Linux or (dare I suggest it) Windows folks.  ;-)

@dsander I made the install blow up several times today - at one point my
network connection was quite busy, and so a *LOT* of gem installs failed.
That iteration was not so fun.  :)

best,

--e

**Comment by cantino** [comment]
I think this can be closed now since we don't have a Vagrantfile.
